segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -754,7 +754,12 @@ void reportMetricsForSystemSegment(SegmentMetadata segmentMetadata) {
      * */
     private <R> CompletableFuture<R> executeSerialized(Callable<CompletableFuture<R>> operation, String... segmentNames) {
         Exceptions.checkNotClosed(this.closed.get(), this);
-        return this.taskProcessor.add(Arrays.asList(segmentNames), () -> executeExclusive(operation, segmentNames));
+        if (segmentNames.length == 1 && this.systemJournal.isStorageSystemSegment(segmentNames[0])) {
+            val segments = this.systemJournal.getSystemSegments();
```

Comment:
To make sure I understand this; if the only segment we have is a system segment, then we linearize on ALL the system segments? This looks a bit counterintuitive - would you mind adding a comment here explaining why so that don't accidentally undo this at some time in the future?

Reply:
Any chance we could write a unit test to verify this? Even with mocking, using `ReusableLatch`, etc.?